### ISSUE TITLE
NullableToStringCall false positive on safe call

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#9074]
## Steps to Reproduce
```kotlin
project.property("…")?.toString()
```
in context:
`if (project.property("…")?.toString().toBoolean())`

## Expected Behavior
No flag.

I understand the original `if (project.property("…").toString().toBoolean())` being flagged (note the missing `?`).

## Observed Behavior
```
NullableToStringCall - [This call 'project.property("…")?.toString()' (...)]
```

It's already compliant because of the `?.toString()` why would I need a `?:` when the `toBoolean` handles that internally?

In addition even this is flagged:
```kotlin
val verbose /* : String? */ = project.property("…")?.toString()
```
where the null stays null, so the NullableToStringCall is not even in effect.

## Context
This is a usual pattern I use in Gradle to check for flags.

## Your Environment
* Version of detekt used: 2.0.0-alpha02
* Version of Gradle used (if applicable): 9.4.0-rc1
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: any
* Link to your project (if it's a public repository):

**Comment by atulgpt** [comment]
Hi @TWiStErRob I tried to run the below TC but it is passing successfully

```kt
@Test
fun `nullable to string on function returning nullable any`() {
    val code = """
        class Project {
            fun property(key: String): Any? = null
        }
        fun foo(project: Project) {
            val verbose /* : String? */ = project.property("…")?.toString()
            if (project.property("…")?.toString().toBoolean()) {
                println("")
            }
        }
    """.trimIndent()
    val actual = subject.lintWithContext(env, code)
    assertThat(actual).isEmpty()
}
```

**Comment by TWiStErRob** [comment]
These things might affect it:

It's defined in Java:
https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/Project.java

And nullability annotations are also involved:
https://github.com/gradle/gradle/blob/master/subprojects/core-api/src/main/java/org/gradle/api/package-info.java

Try it on a real project, not a test case, there's a linked PR where you can hopefully repro it.

**Comment by Phoenix1808** [comment]
Is this issue available to work upon ?

**Comment by cortinico** [comment]
> Is this issue available to work upon ?

Yes but please start by creating a test case where you can reproduce this false positive

**Comment by TWiStErRob** [comment]
@atulgpt do you know of a way to set up java code in the fixture?

**Comment by atulgpt** [comment]
> [@atulgpt](https://github.com/atulgpt) do you know of a way to set up java code in the fixture?

I generally use methods defined in the JDK and call them directly from the test code snippet, so I don’t have to create those methods myself. Off the top of my head, I can’t recall any case where I created a Java code fixture.
